sormas-app/app/src/main/java/de/symeda/sormas/app/backend/contact/ContactDao.java

```diff
@@ -191,53 +195,199 @@ public List<Contact> queryByCriteria(ContactCriteria criteria, long offset, long
+	}
+
+	public Where<Contact, Long> createJurisdictionFilter(Where<Contact, Long> where, ContactCriteria contactCriteria) {
+		List<Where<Contact, Long>> whereJurisdictionFilterStatements = new ArrayList<>();
+
+		User currentUser = ConfigProvider.getUser();
+		if (currentUser == null) {
+			return null;
+		}
+
+		final JurisdictionLevel jurisdictionLevel = currentUser.getJurisdictionLevel();
+
+		if (jurisdictionLevel != JurisdictionLevel.NATION) {
+			//whoever created the case or is assigned to it is allowed to access it
+			if (contactCriteria.getIncludeContactsFromOtherJurisdictions().equals(true)) {
+				whereJurisdictionFilterStatements.add(
+					where.or(
+						where.raw(Case.TABLE_NAME + "." + Case.REPORTING_USER + "= '" + currentUser.getId() + "'"),
+						where.raw(Case.TABLE_NAME + "." + Case.SURVEILLANCE_OFFICER + "= '" + currentUser.getId() + "'"),
+						where.raw(Case.TABLE_NAME + "." + Case.CASE_OFFICER + "= '" + currentUser.getId() + "'")));
+			}
+
+			switch (jurisdictionLevel) {
+			case REGION:
+				Region region = currentUser.getRegion();
+				if (region != null) {
+					whereJurisdictionFilterStatements.add(
+						where.or(
+							where.raw(Case.TABLE_NAME + "." + Case.REGION + "= '" + region + "'"),
+							where.raw(Case.TABLE_NAME + "." + Case.RESPONSIBLE_REGION + "= '" + region.getId() + "'")));
+				}
+				break;
+
+			case DISTRICT:
+				District district = currentUser.getDistrict();
+				if (district != null) {
+					whereJurisdictionFilterStatements.add(
+						where.or(
+							where.raw(Case.TABLE_NAME + "." + Case.DISTRICT + "= '" + district + "'"),
+							where.raw(Case.TABLE_NAME + "." + Case.RESPONSIBLE_DISTRICT + "= '" + district.getId() + "'")));
+				}
+				break;
+
+			case HEALTH_FACILITY:
+				Facility healthFacility = currentUser.getHealthFacility();
+				if (healthFacility != null) {
+					whereJurisdictionFilterStatements.add(where.raw(Case.TABLE_NAME + "." + Case.HEALTH_FACILITY + "= '" + healthFacility + "'"));
+				}
+				break;
+
+			case COMMUNITY:
+				Community community = currentUser.getCommunity();
+				if (community != null) {
+					whereJurisdictionFilterStatements.add(
+						where.or(
+							where.raw(Case.TABLE_NAME + "." + Case.COMMUNITY + "= '" + community + "'"),
+							where.raw(Case.TABLE_NAME + "." + Case.RESPONSIBLE_COMMUNITY + "= '" + community.getId() + "'")));
+				}
+				break;
+
+			case POINT_OF_ENTRY:
+				PointOfEntry pointOfEntry = currentUser.getPointOfEntry();
+				if (pointOfEntry != null) {
+					whereJurisdictionFilterStatements
+						.add(where.raw(Case.TABLE_NAME + "." + Case.POINT_OF_ENTRY + "= '" + pointOfEntry.getId() + "'"));
+				}
+				break;
+			default:
+			}
+		}
+
+		if (!whereJurisdictionFilterStatements.isEmpty()) {
+			where.or(whereJurisdictionFilterStatements.size());
+		}
+		return where;
+	}
+
+	public Where<Contact, Long> createJurisdictionFilterWithoutCase(Where<Contact, Long> where, ContactCriteria contactCriteria) throws SQLException {
```

Comment:
Call this `createJurisdictionFilter` and the other method `createJurisdictionFilterForCase`